t/10-Schema.t

```diff
@@ -37,9 +37,8 @@ lives_ok { $schema->deploy() } 'schema deployed';
     'inner join from run lane to product via custom relationship');
   is ($p_rs->search({'me.id_run' => 4},  {join => 'iseq_run_lane_metric',},)->count(), 4,
     'inner join is performed from product to run lane by default');
-  throws_ok {$p_rs->search({'iseq_run_lane_metric_right.id_run' => 4},  {join => 'iseq_run_lane_metric_right',},)->count()}
```

Comment:
Should we consider dropping this check if switching to newer versions of the test db driver/db libraries? As I understand it, it only applies to sqllite execution.

Reply:
I'm wary of removing tests unless the thing they're testing is covered in another way. The test only fails when using an older version of SQLite. If the tests are run using MySQL, it should pass.

Reply:
It is a difficult call, we are updating this test just to pass CI in github. Unless we make other versions of dependencies (production) match exactly what we use in CI it looks to me we are getting ourselves into trouble. Can the test be made specific? If version of driver and libraries are X or lower run the test, otherwise skip?

Reply:
Just double checked, in our farm devel instance

```
$ TEST_AUTHOR=1 ./Build test --test_files 't/10-Schema.t' --verbose
git version string c7fcb45 not in canonical format, defaulting to 0.0-c7fcb45 at ./Build line 64.
Changing version of all modules and scripts to '0.0-c7fcb45'
t/10-Schema.t .. 
1..10
ok 1 - use WTSI::DNAP::Warehouse::Schema;
ok 2 - connected to test in-memory database
ok 3 - An object of class 'WTSI::DNAP::Warehouse::Schema' isa 'WTSI::DNAP::Warehouse::Schema'
ok 4 - schema deployed
ok 5 - 8 lanes for run 4 created in run-lane table - test prereq.
ok 6 - 4 lanes for run 4 created in product table - test prereq.
ok 7 - left join is performed from run lane to product by default
ok 8 - inner join from run lane to product via custom relationship
ok 9 - inner join is performed from product to run lane by default
DBIx::Class::Storage::DBI::_prepare_sth(): DBI Exception: DBD::SQLite::db prepare_cached failed: RIGHT and FULL OUTER JOINs are not currently supported [for Statement "SELECT COUNT( * ) FROM iseq_product_metrics me RIGHT JOIN iseq_run_lane_metrics iseq_run_lane_metric_right ON ( iseq_run_lane_metric_right.id_run = me.id_run AND iseq_run_lane_metric_right.position = me.position ) WHERE ( iseq_run_lane_metric_right.id_run = ? )"] at t/10-Schema.t line 40
# Looks like your test exited with 255 just after 9.
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/10 subtests 

Test Summary Report
-------------------
t/10-Schema.t (Wstat: 65280 Tests: 9 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 10 tests but ran 9.
Files=1, Tests=9,  5 wallclock secs ( 0.02 usr  0.00 sys +  1.42 cusr  0.16 csys =  1.60 CPU)
Result: FAIL
Failed 1/1 test programs. 0/9 subtests failed.
```
So until we have all versions deployed matching at least the min version which supports this functionality we will have test failing outside of CI. Upgrading sqllite versions in all hosts sounds like out of scope for a change which is CI (github) centric.

Reply:
When you say all versions deployed, where are those deployed versions - Openstack instances?

Is the test valuable? I don't know 100%, but someone thought so originally. In production this uses MySQL, so a better change would be to test against MySQL, IMO. In which case, this problem should go away.

Reply:
It could be internal cloud or compute farms where different (older) versions of sqllite are deployed. The example I gave is from farm.

From [what I understand](https://github.com/wtsi-npg/ml_warehouse/blob/c7fcb45e90e321298fde61d24202b30788fcfa68/t/10-Schema.t#L11) this tests only executes in sqllite. I cannot say why this is only sqllite and not mysql. But I think many of our tests run only on sqllite unless we force mysql for performance. Our unit test is testing our code, not necessarily the platform (mysql or sqllite). I see why this could be controversial.

Regarding value of the test, I think the value goes away once we move to newer versions of sqllite. I'm not sure the test has much value at the moment. We are only testing we get an expected error, but from the code I cannot say why we want that error. The git blame for that commit gives me a [PR](https://github.com/wtsi-npg/ml_warehouse/pull/5) from 8 year ago.

Reply:
Although the test fails only on older SQLite, the comment says it should be working on MySQL. It looks to me like the `throws` test was used to skip over a failure that would ideally have succeeded.